js/pro/timex.js

```diff
@@ -0,0 +1,622 @@
+        //                     },
+        //                     ...
+        //                 ],
+        //             },
+        //         },
+        //     }
+        //  update
+        //     {
+        //         type: "ACCOUNT_SUBSCRIPTION",
+        //         subscriptionId: "95428897-d91a-43e6-9c9e-cebb28fc7b39",
+        //         payload: {
+        //             balance: {
+        //                 currency: "USDT",
+        //                 totalBalance: "16.54435572",
+        //                 lockedBalance: "0",
+        //             },
+        //         },
+        //     }
+        //
+        const payload = this.safeValue (message, 'payload', {});
+        if (this.balance === undefined) {
+            this.balance = {};
+        }
+        let result = this.balance;
+        const snapshot = this.safeValue (payload, 'snapshot', {});
+        const balances = this.safeValue (snapshot, 'balances');
+        if (balances !== undefined) {
+            result = this.parseBalance (balances);
+        }
+        const balance = this.safeValue (payload, 'balance');
+        if (balance !== undefined) {
+            const currencyId = this.safeString (balance, 'currency');
+            const code = this.safeCurrencyCode (currencyId);
+            const account = this.account ();
+            account['total'] = this.safeString (balance, 'totalBalance');
+            account['used'] = this.safeString (balance, 'lockedBalance');
+            result[code] = account;
+        }
+        this.balance = this.safeBalance (result);
+        client.resolve (this.balance, 'balances');
+    }
+
+    async watchTicker (symbol, params = {}) {
+        /**
+         * @method
+         * @name timex#watchTicker
+         * @description watches a price ticker, a statistical calculation with the information calculated over the past 24 hours for a specific market
+         * @url https://docs.timex.io/websocket-api-list-patterns.html#ticker
+         * @param {string} symbol unified symbol of the market to fetch the ticker for
+         * @param {object} params extra parameters specific to the timex api endpoint
+         * @param {string} params.period period of ticker, I1, I5, I15, I30, H1, H2, H4, H6, H12, D1, W1
+         * @returns {object} a [ticker structure]{@link https://docs.ccxt.com/en/latest/manual.html#ticker-structure}
+         */
+        await this.loadMarkets ();
+        const market = this.market (symbol);
+        symbol = this.symbol (symbol);
+        const url = this.urls['api']['ws'];
+        const messageHash = 'ticker:' + symbol;
+        const period = this.safeString (params, 'period', 'D1');
+        params = this.omit (params, period);
+        const message = {
+            'type': 'SUBSCRIBE',
+            'requestId': this.requestId (),
+            'pattern': '/ticker/' + market['id'] + '/' + period,
+        };
+        const request = this.deepExtend (message, params);
```

Comment:
regardless of the period used, this is never returning, do you know why?
![image](https://user-images.githubusercontent.com/43336371/195123399-46fb50a6-47c3-4888-b53a-fe59e2a0e8cb.png)